skills/lark-doc/scripts/count_chars.py

```
@@ -0,0 +1,133 @@
+    # 全角 ASCII 标点（排除全角数字 FF10-FF19、全角字母 FF21-FF3A / FF41-FF5A）
+    if (0xFF01 <= o <= 0xFF0F or 0xFF1A <= o <= 0xFF20
+            or 0xFF3B <= o <= 0xFF40 or 0xFF5B <= o <= 0xFF65
+            or 0xFFE0 <= o <= 0xFFEE):   # 全角货币 ￥￡￠ 等
+        return True
+    return ch in "·—…“”‘’"
+
+
+def count(text):
+    hanzi = sum(1 for ch in text if is_hanzi(ch))
+    zh_punct = sum(1 for ch in text if is_zh_punct(ch))
+    en_words = len(re.findall(r"[A-Za-zÀ-ÿĀ-ɏḀ-ỿ]+", text))
```

Ruff (0.15.20)

[warning] String contains ambiguous `ỿ` (LATIN SMALL LETTER Y WITH LOOP). Did you mean `y` (LATIN SMALL LETTER Y)?

(RUF001)